src/main/kotlin/com/openlattice/indexing/BackgroundIndexingService.kt

```diff
@@ -220,8 +221,14 @@ class BackgroundIndexingService(
 
         val indexCount: Int
 
-        if (entitiesById.isNotEmpty() && elasticsearchApi.createBulkEntityData(entitySet.entityTypeId, entitySet.id, entitiesById)) {
-            indexCount = dataManager.markAsIndexed(mapOf(entitySet.id to Optional.of(batchToIndex)), false)
+        if (entitiesById.isNotEmpty() &&
+                elasticsearchApi.createBulkEntityData(entitySet.entityTypeId, entitySet.id, entitiesById)) {
+            indexCount = if (markAsIndexed) {
+                dataManager.markAsIndexed(mapOf(entitySet.id to Optional.of(batchToIndex)), false)
+            } else {
+                batchToIndex.size
```

Comment:
What is the use case for not marking them as indexed?

Reply:
because this shouldn't interfere with normal background indexing and is designed for maximum throughput (i.e no writes to postgres while getting stuff into ES as fast as possible.

Reply:
Gotcha! Thanks!